docs/user/vsphere/install_upi.md

```diff
@@ -0,0 +1,380 @@
+      ## The password associated with the user
+      password: your_vsphere_password
+      ## The datacenters in the vCenter that will have VMs running OpenShift
+      datacenters:
+      - datacenter_on_your_vsphere_server
+    ## The workspace to use to create VMDK volumes
+    workspace:
+      ## The vSphere server to connect to. This can be omitted when there is only a single vCenter.
+      server: your.vsphere.server
+      ## The datacenter to use. This can be omitted when there is only a single datacenter in the vCenter.
+      datacenter: datacenter_on_your_vsphere_server
+      ## The default datastore to use.
+      defaultDatastore: datastore_on_your_vsphere_server
+      ## The path to the resource pool in which to place volumes.
+      resourcePoolPath: resource_pool_path
+      ## The folder in which VMs running OpenShift reside.
+      folder: vm_folder
+    ## The type of SCSI controller to use
+    scsiControllerType: pvscsi
+    ## The network to which to VMs are attached
+    publicNetwork: VM Network
+## The pull secret that provides components in the cluster access to images for OpenShift components.
+pullSecret: ''
+## The default SSH key that will be programmed for `core` user.
+sshKey: ''
+```
+
+Create a directory that will be used by the OpenShift installer to provide all the assets. For example `test-vsphere`,
+
+```console
+$ mkdir test-vsphere
+$ tree test-vsphere
+test-vsphere
+
+0 directories, 0 files
+```
+
+Copy *your* `install-config` to the `INSTALL_DIR`. For example using the `test-vsphere` as our `INSTALL_DIR`,
+
+```console
+$ cp <your-instal-config> test-vsphere/install-config.yaml
+$ tree test-vsphere
+test-vsphere
+└── install-config.yaml
+
+0 directories, 1 file
+```
+
+NOTE: The filename for `install-config` in the `INSTALL_DIR` must be `install-config.yaml`
+
+### Invoking the installer to get Ignition configs
+
+Given that you have setup the `INSTALL_DIR` with the appropriate `install-config`, you can create the Ignition configs by using the `create ignition-configs` target. For example,
+
+```console
+$ openshift-install --dir test-vsphere create ignition-configs
+INFO Consuming "Install Config" from target directory
+$ tree test-vsphere
+test-vsphere
+├── auth
+│   └── kubeconfig
+├── bootstrap.ign
+├── master.ign
+└── worker.ign
+
+1 directory, 4 files
+```
+
+## Getting OS related assets for machines
+
+TODO RHEL CoreOS does not have assets for vSphere.
+
+## Configuring RHCOS VMs with Ignition configs
+
+Set the vApp properties of the VM to set the Ignition config for the VM. The `guestinfo.ignition.config.data` property is the base64-encoded Ignition config. The `guestinfo.ignition.config.data.encoding` should be set to `base64`.
+
+### Control Plane and Worker VMs
+
+The Ignition configs supplied in the vApp properties of the control plane and worker VMs should be copies of the `master.ign` and `worker.ign` created by the OpenShift Installer.
+
+### Bootstrap VM
```

Comment:
can you provide details on why bootstrap ignition cannot be directly provided to the VMs and why it required indirection.